# This is r2080 from http://serf.googlecode.com/svn/trunk.  Remove
# when an upstream release includes it.
Index: SConstruct
===================================================================
--- SConstruct	(revision 2079)
+++ SConstruct	(revision 2080)
@@ -377,9 +377,13 @@
 install_shared = env.Install(libdir, lib_shared)
 
 if sys.platform == 'darwin':
+  # If --install-sandbox=<path> is specified, install_shared_path will point
+  # to a path in the sandbox. The shared library install name (id) should be the
+  # final targat path.
   install_shared_path = install_shared[0].abspath
+  target_install_shared_path = os.path.join(libdir, lib_shared[0].name)
   env.AddPostAction(install_shared, ('install_name_tool -id %s %s'
-                                     % (install_shared_path,
+                                     % (target_install_shared_path,
                                         install_shared_path)))
   ### construct shared lib symlinks. this also means install the lib
   ### as libserf-2.1.0.0.dylib, then add the symlinks.
